integrations/aws/utils.py

```diff
@@ -0,0 +1,189 @@
+                async with session.client('elbv2') as elbv2:
+                    response = await elbv2.describe_load_balancers(LoadBalancerArns=[identifier])
+                    return response.get('LoadBalancers', [])[0]
+            
+            elif kind == ResourceKindsWithSpecialHandling.CLOUDFORMATION:
+                async with session.client('cloudformation') as cloudformation:
+                    response = await cloudformation.describe_stacks(StackName=identifier)
+                    return response.get('Stacks', [])[0]
+            
+            elif kind == ResourceKindsWithSpecialHandling.EC2:
+                async with session.client('ec2') as ec2_client:
+                    described_instance = await ec2_client.describe_instances(InstanceIds=[identifier])
+                    instance_definition = described_instance["Reservations"][0]["Instances"][0]
+                    return instance_definition
+                
+            else:
+                async with session.client('cloudcontrol') as cloudcontrol:
+                    response = await cloudcontrol.get_resource(TypeName=kind, Identifier=identifier)
+                    resource_description = response.get('ResourceDescription', {})
+                    return {
+                        'Identifier': resource_description.get('Identifier', ''),
+                        'Kind': kind,
+                        **json.loads(resource_description.get('Properties', {}))
+                    }
+        except Exception as e:
+            if e.response['Error']['Code'] == 'ResourceNotFoundException':
+                logger.info(f"Resource not found: {kind} {identifier}")
+                return {}
+            logger.error(f"Failed to describe CloudControl Instance in region: {region}; error {e}")
+            break
+
+async def batch_resources(kind: str, session: aioboto3.Session, service_name: str, describe_method: str, list_param: str, marker_param: str = "NextToken") -> ASYNC_GENERATOR_RESYNC_TYPE:
+    """
+    Describes a list of resources in the AWS account
+    """
+    region = session.region_name
+    account_id = await _session_manager.find_account_id_by_session(session)
+    next_token = None
+    while True:
+        try:
+            all_resources = []
+            async with session.client(service_name) as client:
+                if next_token:
+                    pointer_param = marker_param if marker_param == "NextToken" else "Marker"
+                    response = await getattr(client, describe_method)(**{pointer_param: next_token})
+                else:
+                    response = await getattr(client, describe_method)()
+                next_token = response.get(marker_param)
+                for resource in response.get(list_param, []):
+                    resource.update({KIND_PROPERTY: kind, ACCOUNT_ID_PROPERTY: account_id, REGION_PROPERTY: region, IDENTIFIER_PROPERTY: resource.get('Identifier')})
```

Comment:
Since all the `cloud-resource` can be multiple types of kind it's important to set the kind so the user could filter by kind, it's not given by default

Reply:
lets talk about this i want to see an exmple for the data